pkg/types/v1/common_test.go

```diff
@@ -112,5 +115,95 @@ var _ = Describe("Types", Label("types", "common"), func() {
 			Expect(err).Should(HaveOccurred())
 		})
 	})
+	Describe("Platform", func() {
+		It("Returns the platform", func() {
+			platform, err := v1.NewPlatform("linux", constants.ArchArm64)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(platform.OS).To(Equal("linux"))
+			Expect(platform.Arch).To(Equal(constants.ArchArm64))
+			Expect(platform.GolangArch).To(Equal(constants.ArchArm64))
+			platform, err = v1.NewPlatform("linux", constants.Archx86)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(platform.OS).To(Equal("linux"))
+			Expect(platform.Arch).To(Equal(constants.Archx86))
+			Expect(platform.GolangArch).To(Equal(constants.ArchAmd64))
+		})
+		It("Does not check the validity of the os", func() {
```

Comment:
we currently can only run on linux, no?

Reply:
yes, but its not even checked. Not sure what to do with this, but it may require more validation in the future. For now its just testing whats in there.

Reply:
I kind of remember this being useful to build cross-platform isos/raw images or pull images from other platforms, so it could be used to pull a risc image from a registry for example. So that is why there is no validation and its a free-form thing I think